### PR TITLE
Add support for forward Euler integrator

### PR DESCRIPTION
**Pull request by clinssen**

### CODE REVIEW

File: tests/nest_tests/test_forward_euler_integrator.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+class TestForwardEulerIntegrator:
+    """
+    Tests the forward Euler integrator by comparing it to RK45.
+    """
+
+    def generate_target(self, numeric_solver: str):
+        r"""Generate the neuron model code"""
+
+        # generate the "jit" model (co-generated neuron and synapse), that does not rely on ArchivingNode
```
Comment (pnbabu) [inline tests/nest_tests/test_forward_euler_integrator.py:40]:
```suggestion
```

File: tests/nest_tests/test_forward_euler_integrator.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        nest.Connect(mm1, nrn1)
+        nest.Connect(mm2, nrn2)
+
+        nest.Simulate(100.)
+
+        v_m1 = mm1.get("events")["V_m"]
+        v_m2 = mm2.get("events")["V_m"]
+
+        np.testing.assert_allclose(v_m1, v_m2, atol=2, rtol=0)       # allow max 2 mV difference between the solutions
```
Comment (pnbabu) [inline tests/nest_tests/test_forward_euler_integrator.py:79]: Why 2 mV of difference?

File: pynestml/codegeneration/resources_nest/point_neuron/directives/GSLIntegrationStep.jinja2 [hunk 3]
```diff
@@ -32,3 +33,13 @@ while ( __t < B_.__step )
...
+{%- elif numeric_solver == "forward-Euler" %}
+double f[State_::STATE_VEC_SIZE];
+{{neuronName}}_dynamics( get_t(), S_.ode_state, f, reinterpret_cast< void* >( this ) );
+for (size_t i = 0; i < State_::STATE_VEC_SIZE; ++i)
+{
+    S_.ode_state[i] += __resolution * f[i];
+}
+{%- else %}
+{{ raise('Unknown numeric ODE solver requested.') }}
+{%- endif %}
```
Comment (pnbabu) [inline pynestml/codegeneration/resources_nest/point_neuron/directives/GSLIntegrationStep.jinja2:45]: Add new line

**Review by pnbabu — CHANGES_REQUESTED** [review]

File: tests/nest_tests/test_forward_euler_integrator.py (hunk 2 above)
Comment (clinssen) [reply tests/nest_tests/test_forward_euler_integrator.py:78]: This is as close as we can get with an already very small timestep and the other testing conditions (injected current magnitude, neuron fires a few spikes, etc.)

**Review by pnbabu — APPROVED** [review]
Looks good, thanks!

**Pull request merged**